frontend/src/core/user/models/sessionModel.js

```diff
@@ -67,10 +67,12 @@ define(function(require) {
 
           Origin.trigger('login:failed', errorCode);
         }
+      }).always(function(){
+        self.set('loginFired',false);
```

Comment:
@aniketdharia, I think it would be better to have a more generic approach for this, whether this involves implementing double-click handlers in the views or something else.